From ca238f1f305d5db05a3bde44ae275965765b2ab9 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Fri, 18 Oct 2024 22:40:58 +0200
Subject: [PATCH 2/8] Add IBT support

Rebased for open-gpu-kernel-modules 615.71.09, which added
-mfunction-return=thunk-extern at the end of the x86_64
CONDITIONAL_CFLAGS block. That pushed the "endif" this hunk used as
trailing context down a line, so it stopped applying:

  patching file src/nvidia-modeset/Makefile
  Hunk #1 FAILED at 154.

Signed-off-by: Peter Jung <admin@ptr1337.dev>
---
 src/nvidia-modeset/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/nvidia-modeset/Makefile b/src/nvidia-modeset/Makefile
--- a/src/nvidia-modeset/Makefile
+++ b/src/nvidia-modeset/Makefile
@@ -167,6 +167,7 @@
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch=thunk-extern)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch-register)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mfunction-return=thunk-extern)
+  CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mharden-sls=all)
 endif
 
 CFLAGS += $(CONDITIONAL_CFLAGS)
-- 
2.49.0.391.g4bbb303af6
